homeassistant/components/synology_dsm/__init__.py

```diff
@@ -264,18 +309,12 @@ async def async_unload_entry(hass: HomeAssistantType, entry: ConfigEntry):
 
     if unload_ok:
         entry_data = hass.data[DOMAIN][entry.unique_id]
-        entry_data[UNDO_UPDATE_LISTENER]()
         await entry_data[SYNO_API].async_unload()
         hass.data[DOMAIN].pop(entry.unique_id)
 
     return unload_ok
 
 
-async def _async_update_listener(hass: HomeAssistantType, entry: ConfigEntry):
-    """Handle options update."""
-    await hass.config_entries.async_reload(entry.entry_id)
-
```

Comment:
It looks like the update interval options are no longer going to take effect when they are set since the coordinator update interval is never adjusted when the options are updated.

Reply:
oops, I was probably a little too generous when cleaning up :smiley: